src/test/typescript/io-spec.ts

```diff
@@ -0,0 +1,26 @@
+import { IO } from 'monet';
+
+const logX = IO(() => console.log('X'));
+const getY = logX.map(() => 'Y');
+const getOne = getY.flatMap(y => IO.of(() => y === 'Y' ? 1 : 0));
```

Comment:
This test/example is misleading, because it is an overkill (one can use `map` instead, there are no side-effects here).